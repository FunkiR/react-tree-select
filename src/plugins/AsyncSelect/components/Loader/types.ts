import {ComponentType} from "react";

export type Props = {
	className?: string,
	size: number
};

export type LoaderComponent = ComponentType<Props>;