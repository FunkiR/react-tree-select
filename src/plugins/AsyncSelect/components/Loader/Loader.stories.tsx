import {Meta, Story} from '@storybook/react/types-6-0';
import {withAsyncSelectContext} from "~/plugins/AsyncSelect/stories/decorators";
import React from 'react';

import Loader from './Loader';
import type {Props} from './types';

export default {
	decorators: [withAsyncSelectContext],
	title: 'plugins/AsyncSelect/Components/Loader',
	component: Loader
} as Meta;

const Template: Story<Props> = args => <Loader {...args} />;

export const Default = Template.bind({});

Default.args = {
	size: 25
};

