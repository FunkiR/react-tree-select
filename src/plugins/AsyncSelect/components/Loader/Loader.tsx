import cn from 'classnames';
import useAsyncSelect from "~/plugins/AsyncSelect/hooks/useAsyncSelect";
import React, {FC, memo, useMemo} from 'react';

import type {Props} from './types';

export const Loader: FC<Props> = ({className = '', size}) => {
	const style = useMemo(() => ({height: size, width: size}), [size]);
	const [theme] = useAsyncSelect('theme');

	return <div className={cn(className, theme.Loader)} style={style} />;
};

export default memo(Loader);
