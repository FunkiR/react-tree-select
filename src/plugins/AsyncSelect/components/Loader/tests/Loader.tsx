import {shallow} from "enzyme";
import {Loader} from "~/plugins/AsyncSelect/components/Loader/Loader";
import React from "react";

describe('[AsyncSelect] Loader component', () => {
	it('renders component', () => {
		const wrapper = shallow(<Loader size={25} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('width and height are equal to passed size', () => {
		const size = 25;
		const wrapper = shallow(<Loader size={size} />);
		const style = expect(wrapper.prop('style'));

		style.toHaveProperty('height', size);
		style.toHaveProperty('width', size);
	});
});


