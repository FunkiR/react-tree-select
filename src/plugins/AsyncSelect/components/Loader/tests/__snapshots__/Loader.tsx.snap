// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AsyncSelect] Loader component renders component 1`] = `
<div
  className=""
  style={
    Object {
      "height": 25,
      "width": 25,
    }
  }
/>
`;

exports[`Loader Component renders correctly 1`] = `
<div
  className=""
  style={
    Object {
      "height": 25,
      "width": 25,
    }
  }
/>
`;
